@here/harp-mapview/lib/FrustumIntersection.ts

```diff
@@ -130,6 +133,11 @@ export class FrustumIntersection {
                 continue;
             }
 
+            const proceed = dataSources.some((ds, i) => ds.shouldSubdivide(zoomLevels[i], tileKey));
```

Comment:
Oleksii I guess we could move this section, as also condition above just before uniqueKey getter, this would be even faster, wright? Otherwise looks good!

Reply:
Yes, we can, as well as the section with maxTileLevel above. But I think this unique key is something like sanity check needed for development and beleive it can be completely removed for production build somehow. Anyway we can ping @atomicsulfate for more information about that.

Reply:
The uniqueKey is for wrap around at the antimeridian(date border) and basically just tilekey and offset packed into one number.
It is needed to distinguish the wraped tiles from the "original" tiles.
So please move your code up.